File: Tutorial_Sample_Code/ViewerObjectModel/Parameters/VB/VB_Win_CRVObjMod_Parameters/Form1.vb
Imports CrystalDecisions.CrystalReports.Engine
Imports CrystalDecisions.Shared
Imports System.Collections


Public Class Form1

    Private Const PARAMETER_FIELD_NAME As String = "City"

    Private Sub ConfigureCrystalReports()
        Dim myArrayList As ArrayList = New ArrayList()
        myArrayList.Add("Paris")
        myArrayList.Add("Tokyo")
        Dim reportPath As String = Application.StartupPath & "\" & "CustomersByCity.rpt"
        myCrystalReportViewer.ReportSource = reportPath
        Dim myParameterFields As ParameterFields = myCrystalReportViewer.ParameterFieldInfo
        SetCurrentValuesForParameterField(myParameterFields, myArrayList)
        defaultParameterValuesList.DataSource = GetDefaultValuesFromParameterField(myParameterFields)

    End Sub

    Private Function GetDefaultValuesFromParameterField(ByVal myParameterFields As ParameterFields) As ArrayList
        Dim myParameterField As ParameterField = myParameterFields(PARAMETER_FIELD_NAME)
        Dim defaultParameterValues As ParameterValues = myParameterField.DefaultValues
        Dim myArrayList As ArrayList = New ArrayList()
        For Each myParameterValue As ParameterValue In defaultParameterValues
            If (Not myParameterValue.IsRange) Then
                Dim myParameterDiscreteValue As ParameterDiscreteValue = CType(myParameterValue, ParameterDiscreteValue)
                myArrayList.Add(myParameterDiscreteValue.Value.ToString())
            End If
        Next
        Return myArrayList
    End Function

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        ConfigureCrystalReports()
    End Sub

    Private Sub SetCurrentValuesForParameterField(ByVal myParameterFields As ParameterFields, ByVal myArrayList As ArrayList)
        Dim currentParameterValues As ParameterValues = New ParameterValues()
        For Each submittedValue As Object In myArrayList
            Dim myParameterDiscreteValue As ParameterDiscreteValue = New ParameterDiscreteValue()
            myParameterDiscreteValue.Value = submittedValue.ToString()
            currentParameterValues.Add(myParameterDiscreteValue)
        Next
        Dim myParameterField As ParameterField = myParameterFields(PARAMETER_FIELD_NAME)
        myParameterField.CurrentValues = currentParameterValues
    End Sub

    Private Sub redisplay_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles redisplay.Click
        Dim myArrayList As ArrayList = New ArrayList()
        For Each item As String In defaultParameterValuesList.SelectedItems
            myArrayList.Add(item)
        Next
        myCrystalReportViewer.ReportSource = Application.StartupPath & "\" & "CustomersByCity.rpt"
        Dim myParameterFields As ParameterFields = myCrystalReportViewer.ParameterFieldInfo
        SetCurrentValuesForParameterField(myParameterFields, myArrayList)
    End Sub
End Class

File: Tutorial_Sample_Code/ViewerObjectModel/Parameters/CS/CS_Win_CRVObjMod_Parameters/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using CrystalDecisions.CrystalReports.Engine;
using CrystalDecisions.Shared;
using System.Collections;

namespace CS_Base
{
    public partial class Form1 : Form
    {
        private const string PARAMETER_FIELD_NAME = "City";

        public Form1()
        {
            InitializeComponent();
        }

        private void SetCurrentValuesForParameterField(ParameterFields parameterFields, ArrayList arrayList)
        {
            ParameterValues currentParameterValues = new ParameterValues();
            foreach (object submittedValue in arrayList)
            {
                ParameterDiscreteValue parameterDiscreteValue = new ParameterDiscreteValue();
                parameterDiscreteValue.Value = submittedValue.ToString();
                currentParameterValues.Add(parameterDiscreteValue);
            }
            try
            {
                ParameterField parameterField = parameterFields[PARAMETER_FIELD_NAME];
                parameterField.CurrentValues = currentParameterValues;
            }
            catch
            {
                Console.WriteLine("Exception caught");
            }

        }


        private void ConfigureCrystalReports()
        {
            ArrayList arrayList = new ArrayList();
            arrayList.Add("Paris");
            arrayList.Add("Tokyo");
            string reportPath = Application.StartupPath + "\\" + "CustomersByCity.rpt";
            crystalReportViewer.ReportSource = reportPath;
            ParameterFields parameterFields = crystalReportViewer.ParameterFieldInfo;
            SetCurrentValuesForParameterField(parameterFields, arrayList);
            defaultParameterValuesList.DataSource = GetDefaultValuesFromParameterField(parameterFields);

        }

        private void Form1_Load(object sender, EventArgs e)
        {
            ConfigureCrystalReports();
        }

        private void crystalReportViewer_Load(object sender, EventArgs e)
        {

        }
        private ArrayList GetDefaultValuesFromParameterField(ParameterFields parameterFields)
        {
            ArrayList arrayList = new ArrayList();
            try
            {
                ParameterField parameterField = parameterFields[PARAMETER_FIELD_NAME];
                ParameterValues defaultParameterValues = parameterField.DefaultValues;
                foreach (ParameterValue parameterValue in defaultParameterValues)
                {
                    if (!parameterValue.IsRange)
                    {
                        ParameterDiscreteValue parameterDiscreteValue = (ParameterDiscreteValue)parameterValue;
                        arrayList.Add(parameterDiscreteValue.Value.ToString());
                    }
                }
            }
            catch
            {
                Console.WriteLine("exception is caught");
            }
            return arrayList;
        }

        private void redisplay_Click(object sender, EventArgs e)
        {
            ArrayList arrayList = new ArrayList();
            foreach (string item in defaultParameterValuesList.SelectedItems)
            {
                arrayList.Add(item);
            }
            crystalReportViewer.ReportSource = Application.StartupPath + "\\" + "CustomersByCity.rpt";
            ParameterFields parameterFields = crystalReportViewer.ParameterFieldInfo;
            SetCurrentValuesForParameterField(parameterFields, arrayList);

        }
    }
}